jbi/services.py

```diff
@@ -86,25 +86,51 @@ def jira_visible_projects(jira=None) -> list[dict]:
     return projects
 
 
+class BugzillaClientError(Exception):
+    """Errors raised by `BugzillaClient`."""
+
+
 class BugzillaClient:
-    """
-    Wrapper around the Bugzilla client to turn responses into our models instances.
-    """
+    """A wrapper around `requests` to interact with a Bugzilla REST API."""
 
-    def __init__(self, base_url: str, api_key: str):
-        """Constructor"""
-        self._client = rh_bugzilla.Bugzilla(base_url, api_key=api_key)
+    def __init__(self, base_url, api_key):
+        """Initialize the client, without network activity."""
+        self.base_url = base_url
+        self.params = {
+            "api_key": api_key,
+        }
+        self._client = requests.Session()
```

Comment:
I don't know if this is a better, worse, or neutral change, but we could also do:
```python
    def __init__(self, base_url, api_key):
        """Initialize the client, without network activity."""
        self.base_url = base_url
        _client = requests.Session()
        _client.params = {"api_key": api_key}
        _client.hooks["response"].append(raise_for_status_hook)
        self._client = _client
```

Where `raise_for_status_hook` is 
```python
def raise_for_status_hook(resp: requests.Response, *args, **kwargs):
    resp.raise_for_status()
```

Then `_call` would just be responsible for parsing JSON and error handling:
```python
    def _call(self, verb, url, *args, **kwargs):
        resp = self._client.request(verb, url, *args, **kwargs)
        parsed = resp.json()
        if parsed.get("error"):
            raise BugzillaClientError(parsed["message"])
        return parsed
```

Reply:
I think I prefer having everything in `_call()`. It's not like we have `request.get()` calls everywhere in the code base, the hook approach does not bring much here.
I removed the `self.params` class member to make `_call()` even more confined